### PR TITLE
Correctly handle XTerm.vt100.faceName

### PR DESCRIPTION
**Pull request by aidanharris**
## Description

XTerms font wasn't being detected correctly. This commit fixes that.

### CODE REVIEW

**Comment by aidanharris** [comment]
The following is failing in Travis:

```yaml
# The if statement is here to invert the exit code from grep.
# grep normally errors if no match is found but we want the opposite.
# We invert it so grep fails if a match is found.
- if grep '.\{101\}' neofetch; then (exit 1); else (exit 0); fi
```

`grep '.\{101\}' neofetch` matches the following:

`[ "${term_font:0:1}" != "-" ] && [ "${term_font:0:4}" != "xft:" ] && term_font="xft:$term_font"`

Is there anything wrong with this?

File: neofetch [hunk 1]
```diff
@@ -2183,12 +2183,26 @@ END
...
         "urxvt" | "urxvtd" | "rxvt-unicode" | "xterm")
-            term_font="$(grep -i "${term/d}\**\.*font" < <(xrdb -query))"
+            local _xrdb; _xrdb="$(xrdb -query)"
+            term_font="$(grep -i "${term/d}\**\.*font" <<< "$_xrdb")"
             term_font="${term_font/*"*font:"}"
             term_font="${term_font/*".font:"}"
             term_font="${term_font/*"*.font:"}"
             term_font="$(trim "$term_font")"
 
+            if [ -z "$term_font" ] && [ "$term" = "xterm" ]
```
Comment (dylanaraps) [inline neofetch]:
```sh
if [[ -z "$term_font" && "$term" == "xterm" ]]; then
    # ...
fi
```

File: neofetch [hunk 2]
```diff
@@ -2183,12 +2183,26 @@ END
...
+            then
+                term_font="$(grep -E '^XTerm.vt100.faceName' <<< "$_xrdb")"
+                term_font="${term_font/*"faceName:"}"
+                term_font="$(trim "$term_font")"
+            fi
+
+            # xft: isn't required at the beginning so we append it if it's missing
+            if [ "${term_font:0:1}" != "-" ] && [ "${term_font:0:4}" != "xft:" ]
```
Comment (dylanaraps) [inline neofetch]:
```sh
if [[ "${term_font:0:1}" != "-" && "${term_font:0:4}" != "xft:" ]]; then
    # ...
fi
```

File: neofetch [hunk 3]
```diff
@@ -2183,12 +2183,26 @@ END
         ;;
 
         "urxvt" | "urxvtd" | "rxvt-unicode" | "xterm")
-            term_font="$(grep -i "${term/d}\**\.*font" < <(xrdb -query))"
+            local _xrdb; _xrdb="$(xrdb -query)"
```
Comment (dylanaraps) [inline neofetch]: No need for `local` or the `_var` naming convention. (*I know it's bad but Neofetch uses globals throughout and I want to keep consistency.*)

```sh
xrdb="$(xrdb -query)"
```

**Comment by dylanaraps** [comment]
The `grep` line in the tests checks to see if lines go above 100 chars. You may need to split the if into two lines if it still complains.

**Comment by dylanaraps** [comment]
The if style neofetch uses is `if [[ ]]; then`. Can you update the PR accordingly?

**Comment by aidanharris** [comment]
@dylanaraps Should be good now unless I've missed something.

**Comment by dylanaraps** [comment]
Just fix the `if` style and I'll merge. :+1:

**Comment by aidanharris** [comment]
@dylanaraps I've fixed the errors now and force pushed so this should be ready to be merged.

File: neofetch [hunk 4]
```diff
@@ -2183,12 +2183,25 @@ END
...
             term_font="$(trim "$term_font")"
 
+            if [[ -z "$term_font" && "$term" == "xterm" ]]; then
+                term_font="$(grep -E '^XTerm.vt100.faceName' <<< "$xrdb")"
+                term_font="${term_font/*"faceName:"}"
+                term_font="$(trim "$term_font")"
+            fi
+
+            # xft: isn't required at the beginning so we prepend it if it's missing
+            if [[ "${term_font:0:1}" != "-" ]] && \
```
Comment (dylanaraps) [inline neofetch]: Omit the second set of braces.

```sh
            if [[ "${term_font:0:1}" != "-" && \
                  "${term_font:0:4}" != "xft:" ]]; then
```

Comment (aidanharris) [reply neofetch]: Sorry about that. That be fixed now.

File: neofetch [hunk 5]
```diff
@@ -2183,12 +2183,25 @@ END
...
+            term_font="$(grep -i "${term/d}\**\.*font" <<< "$xrdb")"
             term_font="${term_font/*"*font:"}"
             term_font="${term_font/*".font:"}"
             term_font="${term_font/*"*.font:"}"
             term_font="$(trim "$term_font")"
 
+            if [[ -z "$term_font" && "$term" == "xterm" ]]; then
+                term_font="$(grep -E '^XTerm.vt100.faceName' <<< "$xrdb")"
+                term_font="${term_font/*"faceName:"}"
+                term_font="$(trim "$term_font")"
```
Comment (dylanaraps) [inline neofetch]: `trim` is probably unneeded. Every string gets trimmed before printing anyway. Unless you've got a reason for keeping it?

**Comment by dylanaraps** [comment]
Sorry for being so anal :P

**Pull request merged**